libs/libskynet/src/err.ts

```diff
@@ -0,0 +1,47 @@
+import { tryStringify } from "./stringifytry.js";
+
+// addContextToErr is a helper function that standardizes the formatting of
+// adding context to an error. Within the world of go we discovered that being
+// persistent about layering context onto errors is helpful when debugging,
+// even though it often creates rather verbose error messages.
+//
+// addContextToErr will return null if the input err is null.
+//
+// NOTE: To protect against accidental situations where an Error type or some
+// other type is provided instead of a string, we wrap both of the inputs with
+// tryStringify before returning them. This prevents runtime failures.
+function addContextToErr(err: any, context: string): string {
+  if (err === null) {
```

Comment:
Do we want to check for `undefined` as well? `if (err == null)` would do just that.

Reply:
hmm the type here should probably be `string | null` instead of `any`. I think I made it `any` so that it could be used in `catch` blocks. Maybe we should leave this function general and also check for `undefined`.